webhook/resources/volume/validator.go

```diff
@@ -156,6 +164,53 @@ func (v *volumeValidator) Update(request *admission.Request, oldObj runtime.Obje
 	return nil
 }
 
+func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newVolume *longhorn.Volume) error {
+	oldSize := oldVolume.Spec.Size
+	newSize := newVolume.Spec.Size
+	if newSize == oldSize {
+		return nil
+	}
+	if newSize < oldSize && !newVolume.Status.ExpansionRequired {
+		return fmt.Errorf("shrinking volume size from %v/%v to %v/%v is not supported", oldVolume.Name, oldSize, newVolume.Name, newSize)
+	}
+
+	newKubernetesStatus := &newVolume.Status.KubernetesStatus
+	namespace := newKubernetesStatus.Namespace
+	pvcName := newKubernetesStatus.PVCName
+	if pvcName == "" || newKubernetesStatus.LastPVCRefAt != "" {
+		return nil
+	}
+
+	pvc, err := v.ds.GetPersistentVolumeClaim(namespace, pvcName)
+	if err != nil {
+		return err
+	}
+
+	pvcSCName := *pvc.Spec.StorageClassName
+	pvcStorageClass, err := v.ds.GetStorageClassRO(pvcSCName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "cannot find storage class %v", pvcSCName)
+		}
+		return err
+	}
+	if pvcStorageClass.AllowVolumeExpansion != nil && !*pvcStorageClass.AllowVolumeExpansion {
+		return fmt.Errorf("storage class does not allow the volume expansion")
```

Comment:
nit: should add volume info like name and storage class name in the error message like what you did above.

Reply:
Added the pvcSCName and pvc.Name into this error log. Thanks